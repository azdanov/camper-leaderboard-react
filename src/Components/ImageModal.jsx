// @flow
import * as React from 'react';
import { Image, Modal } from 'semantic-ui-react';

const ImageModal = (props: { img: string, username: string }) => (
  <Modal
    basic
    closeIcon
    size="mini"
    trigger={<Image className="clickable" src={props.img} size="huge" spaced />}
  >
    <Modal.Header>{props.username}</Modal.Header>
    <Modal.Content image>
      <Image wrapped size="medium" src={props.img} />
    </Modal.Content>
  </Modal>
);

export default ImageModal;
