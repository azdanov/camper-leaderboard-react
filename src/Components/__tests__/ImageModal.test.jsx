// @flow

import React from 'react';

import { shallow } from 'enzyme';
import ImageModal from '../ImageModal';

test('Head renders correctly', () => {
  const component = shallow(<ImageModal username="test" img="linktoimage" />);
  expect(component).toMatchSnapshot();
});
