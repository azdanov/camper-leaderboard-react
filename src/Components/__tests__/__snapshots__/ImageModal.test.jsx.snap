// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Modal
    basic={true}
    closeIcon={true}
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    size="mini"
    trigger={
      <Image
        as="img"
        className="clickable"
        size="huge"
        spaced={true}
        src="linktoimage"
        ui={true}
      />
    }
  >
    <ModalHeader>
      test
    </ModalHeader>
    <ModalContent
      image={true}
    >
      <Image
        as="img"
        size="medium"
        src="linktoimage"
        ui={true}
        wrapped={true}
      />
    </ModalContent>
  </Modal>,
  "nodes": Array [
    <Modal
      basic={true}
      closeIcon={true}
      closeOnDimmerClick={true}
      closeOnDocumentClick={false}
      dimmer={true}
      size="mini"
      trigger={
        <Image
          as="img"
          className="clickable"
          size="huge"
          spaced={true}
          src="linktoimage"
          ui={true}
        />
      }
    >
      <ModalHeader>
        test
      </ModalHeader>
      <ModalContent
        image={true}
      >
        <Image
          as="img"
          size="medium"
          src="linktoimage"
          ui={true}
          wrapped={true}
        />
      </ModalContent>
    </Modal>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <ImageModal
        img="linktoimage"
        username="test"
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "img": "linktoimage",
          "username": "test",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Modal
          basic={true}
          closeIcon={true}
          closeOnDimmerClick={true}
          closeOnDocumentClick={false}
          dimmer={true}
          size="mini"
          trigger={
            <Image
              as="img"
              className="clickable"
              size="huge"
              spaced={true}
              src="linktoimage"
              ui={true}
            />
          }
        >
          <ModalHeader>
            test
          </ModalHeader>
          <ModalContent
            image={true}
          >
            <Image
              as="img"
              size="medium"
              src="linktoimage"
              ui={true}
              wrapped={true}
            />
          </ModalContent>
        </Modal>,
        "_debugID": 2,
        "_renderedOutput": <Modal
          basic={true}
          closeIcon={true}
          closeOnDimmerClick={true}
          closeOnDocumentClick={false}
          dimmer={true}
          size="mini"
          trigger={
            <Image
              as="img"
              className="clickable"
              size="huge"
              spaced={true}
              src="linktoimage"
              ui={true}
            />
          }
        >
          <ModalHeader>
            test
          </ModalHeader>
          <ModalContent
            image={true}
          >
            <Image
              as="img"
              size="medium"
              src="linktoimage"
              ui={true}
              wrapped={true}
            />
          </ModalContent>
        </Modal>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <ImageModal
    img="linktoimage"
    username="test"
  />,
}
`;
